### PR TITLE
Feature/add report

### PR DESCRIPTION
**Pull request by andresgnlez**
Pivotal task: https://www.pivotaltracker.com/story/show/142962043

This PR adds:

- Add/Remove indicators functionality in localStorage.
- Adds a fixed bar that counts the added indicators.

### CODE REVIEW

File: app/assets/javascripts/components/data_portal/modals/modal-save-widget.js [hunk 1]
```diff
@@ -79,6 +168,38 @@
...
+      this.el.querySelector('.js-remove-report').classList.toggle('_is-hidden', !isSaved);
+    },
+
+    _onAddReport: function () {
+      var indicator = _.extend({}, this.options.widgetConfig);
+
+      // adds indicator to localStorage
+      this._saveIndicator(indicator);
+
+      // disables 'Add to report' button once the widget is added
```
Comment (clementprdhomme) [inline app/assets/javascripts/components/data_portal/modals/modal-save-widget.js]: This code is the same as the one in `this._toggleButtons`.

File: app/assets/javascripts/components/data_portal/modals/modal-save-widget.js [hunk 2]
```diff
@@ -79,6 +168,38 @@
...
+      this.el.querySelector('.js-remove-report').classList.remove('_is-hidden');
+    },
+
+    _onRemoveReport: function () {
+      var indicator = _.extend({}, this.options.widgetConfig);
+
+      // removes indicator from localStorage
+      this._removeIndicator(indicator);
+
+      this.el.querySelector('.js-add-report').classList.remove('_is-hidden');
```
Comment (clementprdhomme) [inline app/assets/javascripts/components/data_portal/modals/modal-save-widget.js]: Same here

File: app/assets/javascripts/views/data_portal/ReportFixedBarView.js [hunk 3]
```diff
@@ -0,0 +1,45 @@
...
+    },
+
+    render: function () {
+      var widgetsOnReport = localStorage.getItem('indicators') ? JSON.parse(localStorage.getItem('indicators')).length : null;
+
+      this.el.classList.toggle('_is-hidden', !widgetsOnReport);
+
+      $('body').append($(this.el).html(
+        this.template({
+          encondedIndicators: this._encodeIndicators(),
```
Comment (clementprdhomme) [inline app/assets/javascripts/views/data_portal/ReportFixedBarView.js]: "encoded" not "enconded" 😉

File: app/assets/javascripts/components/data_portal/modals/modal-save-widget.js [hunk 4]
```diff
@@ -43,28 +44,116 @@
       }];
     },
 
-    _setPosition: function () {
+    //  *** localStorage private methods ***
+
+    /**
+     * @return array - Returns an array of indicators or an empty array
```
Comment (clementprdhomme) [inline app/assets/javascripts/components/data_portal/modals/modal-save-widget.js]: It's a good thing you've started to add comments, but you should check the [proper syntax](http://usejsdoc.org/about-getting-started.html).

After `@return` or `@param`, you must write the type of the variable between braces and then the name of the param in the case of `@param`.

Note that `@params` doesn't exist, it's always `@param`.

Comment (clementprdhomme) [inline app/assets/javascripts/components/data_portal/modals/modal-save-widget.js]: BTW, `array` is not a type. In this case it's `object[]` or `{}[]`.

Comment (andresgnlez) [inline app/assets/javascripts/components/data_portal/modals/modal-save-widget.js]: Thanks for the link! 😁

File: app/assets/javascripts/views/data_portal/ReportFixedBarView.js (hunk 3 above)
Comment (andresgnlez) [inline app/assets/javascripts/views/data_portal/ReportFixedBarView.js]: karma 😝

File: app/assets/javascripts/components/data_portal/modals/modal-save-widget.js [hunk 5]
```diff
@@ -43,28 +44,117 @@
       }];
     },
 
-    _setPosition: function () {
+    //  *** localStorage private methods ***
+
+    /**
+     * Returns an array of indicators or an empty array
+     * @return object[] - Array of indicators
```
Comment (clementprdhomme) [inline app/assets/javascripts/components/data_portal/modals/modal-save-widget.js]: The curly braces are missing around `object[]`: `@return {object[]} ...`

File: app/assets/javascripts/views/data_portal/ReportFixedBarView.js [hunk 6]
```diff
@@ -0,0 +1,45 @@
...
+      this.render();
+    },
+
+    _setEventListeners: function () {
+      Backbone.Events.on('indicator:saved', this.render.bind(this));
+    },
+
+    /**
+     * Encondes saved indicators array to b64. More info in helpers/url
+     * @return base64 string of saved indicator array
```
Comment (clementprdhomme) [inline app/assets/javascripts/views/data_portal/ReportFixedBarView.js]: `@return {string} base64 string ...`

**Pull request merged**